### PR TITLE
Replace pytest's paremeterization with parameterized 

### PR DESCRIPTION
**Pull request by krishnakalyan3**
Made Changes to

- [x] test/torchaudio_unittest/librosa_compatibility_test.py
- [x] test/torchaudio_unittest/functional/functional_cpu_test.py

fixes: #1156

### CODE REVIEW

File: test/torchaudio_unittest/functional/functional_cpu_test.py [hunk 1]
```diff
@@ -121,16 +121,18 @@ def test_DB_to_amplitude(self):
...
-])
-@pytest.mark.parametrize('power', [1, 2, 0.7])
-def test_complex_norm(complex_tensor, power):
-    expected_norm_tensor = complex_tensor.pow(2).sum(-1).pow(power / 2)
-    norm_tensor = F.complex_norm(complex_tensor, power)
-
-    torch.testing.assert_allclose(norm_tensor, expected_norm_tensor, atol=1e-5, rtol=1e-5)
+class TestComplexNorm(common_utils.TorchaudioTestCase):
+
+    @parameterized.expand([
```
Comment (krishnakalyan3) [inline test/torchaudio_unittest/functional/functional_cpu_test.py:126]: I suppose this is not quite right. `list(itertools.product(...)` might be the right way to go about this. However that would require importing `import itertools`. 

@mthrok which would be the best way to go about this?.

**Comment by krishnakalyan3** [comment]
Following occurrences will be substituted (`pytest.mark.parametrize -> parameterized`)  eventually as part of this PR.

image

File: test/torchaudio_unittest/functional/functional_cpu_test.py (hunk 1 above)
Comment (mthrok) [reply test/torchaudio_unittest/functional/functional_cpu_test.py:126]: Yeah, that's right. That's the pattern I use everywhere. `itertools` is a standard library so there is no problem.
Maybe it's good time to add a helper decorator for `@partameterized.expand(list(itertools.product` pattern, but let's not bother at this moment.

File: test/torchaudio_unittest/functional/functional_cpu_test.py [hunk 2]
```diff
@@ -121,16 +121,18 @@ def test_DB_to_amplitude(self):
...
+        (torch.randn(1, 2, 1025, 400, 2),  1),
+        (torch.randn(1025, 400, 2), 2)
+    ])
+    def test_complex_norm(self, complex_tensor, power):
```
Comment (mthrok) [inline test/torchaudio_unittest/functional/functional_cpu_test.py:130]: Performing `torch.randn` in decorator makes it difficult to control the randomness, so instead can you parameterize the shape?

```
def test_complex_norm(self, shape, power):
    torch.manual_seed(0)
    complex_tensor = torch.randn(*shape)
```

File: test/torchaudio_unittest/functional/functional_cpu_test.py [hunk 3]
```diff
@@ -121,52 +123,57 @@ def test_DB_to_amplitude(self):
...
+    @parameterized.expand(list(itertools.product(
+        [(1, 2, 1025, 400, 2), (1025, 400, 2)],
+        [1, 2, 0.7]
+    )))
+    def test_complex_norm(self, shape, power):
+        torch.random.manual_seed(42)
+        complex_tensor = torch.randn(*shape)
+        expected_norm_tensor = complex_tensor.pow(2).sum(-1).pow(power / 2)
+        norm_tensor = F.complex_norm(complex_tensor, power)
+        torch.testing.assert_allclose(norm_tensor, expected_norm_tensor, atol=1e-5, rtol=1e-5)
```
Comment (mthrok) [inline test/torchaudio_unittest/functional/functional_cpu_test.py:136]: Can you replace `torch.testing.assert_allclose` with `self.assertEqual`?

the use of `torch.testing.assert_allclose` is discouraged now.

File: test/torchaudio_unittest/functional/functional_cpu_test.py [hunk 4]
```diff
@@ -81,6 +82,7 @@ def test_pitch(self, frequency):
 
 class TestDB_to_amplitude(common_utils.TorchaudioTestCase):
     def test_DB_to_amplitude(self):
+        torch.random.manual_seed(42)
```
Comment (mthrok) [inline test/torchaudio_unittest/functional/functional_cpu_test.py:85]: Let's leave this test untouched. #1113 is modifying this test suite.

File: test/torchaudio_unittest/librosa_compatibility_test.py [hunk 5]
```diff
@@ -111,42 +113,49 @@ def test_amplitude_to_DB(self):
...
+            mono_complex_specgram[..., 1] * 1j
+        expected_complex_stretch = librosa.phase_vocoder(
+            mono_complex_specgram,
+            rate=rate,
+            hop_length=hop_length)
+
+        complex_stretch = complex_specgrams_stretch[index].numpy()
+        complex_stretch = complex_stretch[..., 0] + 1j * complex_stretch[..., 1]
+
+        assert np.allclose(complex_stretch, expected_complex_stretch, atol=1e-5)
```
Comment (mthrok) [inline test/torchaudio_unittest/librosa_compatibility_test.py:158]: Can you use `self.assertEqual` here? You might need to wrap the expected using `torch.from_numpy`.

**Review by mthrok — COMMENTED** [review]
Looks good. Made some suggestions.

**Review by mthrok — APPROVED** [review]

**Comment by mthrok** [comment]
Thanks!

**Pull request merged**